robottelo/hosts.py

```diff
@@ -1169,35 +1169,39 @@ def register_to_dogfood(self, ak_type='satellite'):
 
     def capsule_setup(self, sat_host=None, **installer_kwargs):
         """Prepare the host and run the capsule installer"""
-        if sat_host is not None:
-            self._satellite = sat_host
-        self.register_to_dogfood(ak_type='capsule')
-        self._satellite = sat_host or Satellite()
-
-        # self.execute('yum repolist')
-        self.execute('yum -y update', timeout=0)
-
-        if settings.server.version.rhel_version == 8:
+        # Register capsule host to CDN and enable repos
+        self.register_contenthost(
+            org=None,
+            lce=None,
+            username=settings.subscription.rhn_username,
+            password=settings.subscription.rhn_password,
+            auto_attach=True,
+        )
+        if settings.server.version.rhel_version == 7:
+            self.enable_repo(repo=constants.REPOS['rhscl7']['id'])
+            self.enable_repo(repo=constants.REPOS['rhae2']['id'])
```

Comment:
You would need too `self.enable_repo(repo=constants.REPOS['rhel7']['id'])` ?

Reply:
I noticed it's enabled automatically after register/attach. But you are right it makes no harm to have it explicitly stated here.